.github/workflows/k8s_e2e.yml

```diff
@@ -76,8 +81,9 @@ jobs:
   build-x86_64-unknown-linux-gnu:
     name: Build - x86_64-unknown-linux-gnu
     runs-on: [linux, ubuntu-20.04-4core]
-    needs: [changes, validate]
-    if: github.event_name != 'pull_request' || needs.changes.outputs.k8s == 'true'
+    needs: changes
+    # Run this job even if `changes` job is skipped (non- pull request trigger)
+    if: always() && (github.event_name != 'pull_request' || needs.changes.outputs.k8s == 'true')
```

Comment:
Will adding `always()` here cause this step to run even when the "Validate" job is skipped? For instance, would this now run even when a comment without `/ci-run-all` or `ci-run-k8s` is created?

Reply:
This is now irrelevant because the comment body detection is all in the caller workflow. But you were right, this would have had to have the other conditional repeated here.

Reply:
We should try to avoid using `always()` because we shouldn't continue execution if `changes` fails and it prevents us from being able to cancel the workflow.  Due to various historical reasons ([ref](https://github.com/actions/runner/issues/491#issuecomment-926924523)), I believe we can use `!failure() && !cancelled()` to check if the previous jobs were successful or skipped. It's ugly but looks to be the best option available, as there is also `success()`, but sadly no `skipped()`.

```suggestion
    # Run this job even if `changes` job is skipped (non- pull request trigger)
    if: !failure() && !cancelled() && (github.event_name != 'pull_request' || needs.changes.outputs.k8s == 'true')
```